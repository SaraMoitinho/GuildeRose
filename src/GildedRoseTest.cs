using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using System;
using System.Collections.Generic;
using Xunit;

namespace MeuAcerto.Selecao.KataGildedRose
{
  [UseReporter(typeof(DiffReporter))]
  public class GildedRoseTest
  {
    [Fact]
    public void DadoQueAtualizaQualidade()
    {
     
      CombinationApprovals.VerifyAllCombinations(AtualizarQualidade,
        new String[] {"foo", "Queijo Brie Envelhecido", "Ingressos para o concerto do Turisas", "Dente do Tarrasque", "Bolo de Mana Conjurado" },
        new int[] {-1, 0, 2, 6, 11},
        new int[] {0, 1, 49, 50});
    }

    private static string AtualizarQualidade(string nomeItem, int valorVenda, int valorQualidade)
    {
      List<Item> itens = new List<Item>
        {

        new Item(nomeItem,valorVenda, valorQualidade).CriarItem()
      };

      GildedRose app = new GildedRose(itens);
      app.AtualizarQualidade();
      return app.Itens[0].ObterPropriedades();
    }
  }
}
