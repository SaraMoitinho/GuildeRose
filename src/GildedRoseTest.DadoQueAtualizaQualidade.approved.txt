[foo,-1,0] => foo, -2, 0
[foo,-1,1] => foo, -2, 0
[foo,-1,49] => foo, -2, 47
[foo,-1,50] => foo, -2, 48
[foo,0,0] => foo, -1, 0
[foo,0,1] => foo, -1, 0
[foo,0,49] => foo, -1, 47
[foo,0,50] => foo, -1, 48
[foo,2,0] => foo, 1, 0
[foo,2,1] => foo, 1, 0
[foo,2,49] => foo, 1, 48
[foo,2,50] => foo, 1, 49
[foo,6,0] => foo, 5, 0
[foo,6,1] => foo, 5, 0
[foo,6,49] => foo, 5, 48
[foo,6,50] => foo, 5, 49
[foo,11,0] => foo, 10, 0
[foo,11,1] => foo, 10, 0
[foo,11,49] => foo, 10, 48
[foo,11,50] => foo, 10, 49
[Queijo Brie Envelhecido,-1,0] => Queijo Brie Envelhecido, -2, 2
[Queijo Brie Envelhecido,-1,1] => Queijo Brie Envelhecido, -2, 3
[Queijo Brie Envelhecido,-1,49] => Queijo Brie Envelhecido, -2, 50
[Queijo Brie Envelhecido,-1,50] => Queijo Brie Envelhecido, -2, 50
[Queijo Brie Envelhecido,0,0] => Queijo Brie Envelhecido, -1, 2
[Queijo Brie Envelhecido,0,1] => Queijo Brie Envelhecido, -1, 3
[Queijo Brie Envelhecido,0,49] => Queijo Brie Envelhecido, -1, 50
[Queijo Brie Envelhecido,0,50] => Queijo Brie Envelhecido, -1, 50
[Queijo Brie Envelhecido,2,0] => Queijo Brie Envelhecido, 1, 1
[Queijo Brie Envelhecido,2,1] => Queijo Brie Envelhecido, 1, 2
[Queijo Brie Envelhecido,2,49] => Queijo Brie Envelhecido, 1, 50
[Queijo Brie Envelhecido,2,50] => Queijo Brie Envelhecido, 1, 50
[Queijo Brie Envelhecido,6,0] => Queijo Brie Envelhecido, 5, 1
[Queijo Brie Envelhecido,6,1] => Queijo Brie Envelhecido, 5, 2
[Queijo Brie Envelhecido,6,49] => Queijo Brie Envelhecido, 5, 50
[Queijo Brie Envelhecido,6,50] => Queijo Brie Envelhecido, 5, 50
[Queijo Brie Envelhecido,11,0] => Queijo Brie Envelhecido, 10, 1
[Queijo Brie Envelhecido,11,1] => Queijo Brie Envelhecido, 10, 2
[Queijo Brie Envelhecido,11,49] => Queijo Brie Envelhecido, 10, 50
[Queijo Brie Envelhecido,11,50] => Queijo Brie Envelhecido, 10, 50
[Ingressos para o concerto do Turisas,-1,0] => Ingressos para o concerto do Turisas, -2, 0
[Ingressos para o concerto do Turisas,-1,1] => Ingressos para o concerto do Turisas, -2, 0
[Ingressos para o concerto do Turisas,-1,49] => Ingressos para o concerto do Turisas, -2, 0
[Ingressos para o concerto do Turisas,-1,50] => Ingressos para o concerto do Turisas, -2, 0
[Ingressos para o concerto do Turisas,0,0] => Ingressos para o concerto do Turisas, -1, 0
[Ingressos para o concerto do Turisas,0,1] => Ingressos para o concerto do Turisas, -1, 0
[Ingressos para o concerto do Turisas,0,49] => Ingressos para o concerto do Turisas, -1, 0
[Ingressos para o concerto do Turisas,0,50] => Ingressos para o concerto do Turisas, -1, 0
[Ingressos para o concerto do Turisas,2,0] => Ingressos para o concerto do Turisas, 1, 3
[Ingressos para o concerto do Turisas,2,1] => Ingressos para o concerto do Turisas, 1, 4
[Ingressos para o concerto do Turisas,2,49] => Ingressos para o concerto do Turisas, 1, 50
[Ingressos para o concerto do Turisas,2,50] => Ingressos para o concerto do Turisas, 1, 50
[Ingressos para o concerto do Turisas,6,0] => Ingressos para o concerto do Turisas, 5, 2
[Ingressos para o concerto do Turisas,6,1] => Ingressos para o concerto do Turisas, 5, 3
[Ingressos para o concerto do Turisas,6,49] => Ingressos para o concerto do Turisas, 5, 50
[Ingressos para o concerto do Turisas,6,50] => Ingressos para o concerto do Turisas, 5, 50
[Ingressos para o concerto do Turisas,11,0] => Ingressos para o concerto do Turisas, 10, 1
[Ingressos para o concerto do Turisas,11,1] => Ingressos para o concerto do Turisas, 10, 2
[Ingressos para o concerto do Turisas,11,49] => Ingressos para o concerto do Turisas, 10, 50
[Ingressos para o concerto do Turisas,11,50] => Ingressos para o concerto do Turisas, 10, 50
[Dente do Tarrasque,-1,0] => Dente do Tarrasque, -1, 0
[Dente do Tarrasque,-1,1] => Dente do Tarrasque, -1, 1
[Dente do Tarrasque,-1,49] => Dente do Tarrasque, -1, 49
[Dente do Tarrasque,-1,50] => Dente do Tarrasque, -1, 50
[Dente do Tarrasque,0,0] => Dente do Tarrasque, 0, 0
[Dente do Tarrasque,0,1] => Dente do Tarrasque, 0, 1
[Dente do Tarrasque,0,49] => Dente do Tarrasque, 0, 49
[Dente do Tarrasque,0,50] => Dente do Tarrasque, 0, 50
[Dente do Tarrasque,2,0] => Dente do Tarrasque, 2, 0
[Dente do Tarrasque,2,1] => Dente do Tarrasque, 2, 1
[Dente do Tarrasque,2,49] => Dente do Tarrasque, 2, 49
[Dente do Tarrasque,2,50] => Dente do Tarrasque, 2, 50
[Dente do Tarrasque,6,0] => Dente do Tarrasque, 6, 0
[Dente do Tarrasque,6,1] => Dente do Tarrasque, 6, 1
[Dente do Tarrasque,6,49] => Dente do Tarrasque, 6, 49
[Dente do Tarrasque,6,50] => Dente do Tarrasque, 6, 50
[Dente do Tarrasque,11,0] => Dente do Tarrasque, 11, 0
[Dente do Tarrasque,11,1] => Dente do Tarrasque, 11, 1
[Dente do Tarrasque,11,49] => Dente do Tarrasque, 11, 49
[Dente do Tarrasque,11,50] => Dente do Tarrasque, 11, 50
[Bolo de Mana Conjurado,-1,0] => Bolo de Mana Conjurado, -2, 0
[Bolo de Mana Conjurado,-1,1] => Bolo de Mana Conjurado, -2, 0
[Bolo de Mana Conjurado,-1,49] => Bolo de Mana Conjurado, -2, 0
[Bolo de Mana Conjurado,-1,50] => Bolo de Mana Conjurado, -2, 0
[Bolo de Mana Conjurado,0,0] => Bolo de Mana Conjurado, -1, 0
[Bolo de Mana Conjurado,0,1] => Bolo de Mana Conjurado, -1, 0
[Bolo de Mana Conjurado,0,49] => Bolo de Mana Conjurado, -1, 0
[Bolo de Mana Conjurado,0,50] => Bolo de Mana Conjurado, -1, 0
[Bolo de Mana Conjurado,2,0] => Bolo de Mana Conjurado, 1, 0
[Bolo de Mana Conjurado,2,1] => Bolo de Mana Conjurado, 1, 0
[Bolo de Mana Conjurado,2,49] => Bolo de Mana Conjurado, 1, 47
[Bolo de Mana Conjurado,2,50] => Bolo de Mana Conjurado, 1, 48
[Bolo de Mana Conjurado,6,0] => Bolo de Mana Conjurado, 5, 0
[Bolo de Mana Conjurado,6,1] => Bolo de Mana Conjurado, 5, 0
[Bolo de Mana Conjurado,6,49] => Bolo de Mana Conjurado, 5, 47
[Bolo de Mana Conjurado,6,50] => Bolo de Mana Conjurado, 5, 48
[Bolo de Mana Conjurado,11,0] => Bolo de Mana Conjurado, 10, 0
[Bolo de Mana Conjurado,11,1] => Bolo de Mana Conjurado, 10, 0
[Bolo de Mana Conjurado,11,49] => Bolo de Mana Conjurado, 10, 47
[Bolo de Mana Conjurado,11,50] => Bolo de Mana Conjurado, 10, 48
